xarray/core/dataarray.py

```diff
@@ -1369,6 +1369,83 @@ def real(self):
     @property
     def imag(self):
         return self._replace(self.variable.imag)
+    
+    def tensordot( self, b, dims):
+        a = self
+        if not (isinstance(a, DataArray) and isinstance(b, DataArray)):
+            raise ValueError
+    
+        a, b = align(a, b, join='inner', copy=False)
+    
+        axes = (a.get_axis_num(dims), b.get_axis_num(dims))
+        f = ops._dask_or_eager_func('tensordot', n_array_args=2)
+        new_data = f(a.data, b.data, axes=axes)
+    
+        if isinstance(dims, str):
+            dims = [dims]
+    
+        new_coords = a.coords.merge(b.coords).drop(dims)
+    
+        new_dims = ([d for d in a.dims if d not in dims] +
+                    [d for d in b.dims if d not in dims])
+    
+        return DataArray(new_data, new_coords, new_dims)
+
+    def dot(self, other):
+        """Perform dot product of two DataArrays along their shared dims.
+        
+        Equivalent to taking taking tensordot over all shared dims.
+
+        Parameters
+        ----------
+        other : DataArray
+            The other array with which the dot product is performed.
+            
+        Returns
+        -------
+        result : DataArray
+            Array resulting from the dot product over all shared dimensions.
+            
+        See also
+        --------
+        np.tensordot(a, b, axes)
+
+        Examples
+        --------
+
+        >>> da_vals = np.arange(6*5*4).reshape((6, 5, 4))
```

Comment:
This still has the wrong spacing between numbers per PEP8 -- it should be `6 * 5 * 4`.

I would also consider making a slightly simpler example, e.g., a tensor with dimensions `(x, y, z)` dot `(z,)` -> `(x, y)`. This is probably fine, though